recipe/test/CMakeLists.txt

```diff
@@ -6,13 +6,19 @@ cmake_minimum_required(VERSION 3.1)
 project(test_cgal)
 find_package(Boost REQUIRED COMPONENTS thread)
 find_package(CGAL REQUIRED COMPONENTS Core)
+
+# Eigen3 was added to the test in order to ensure that FindEigen3.cmake from
+# inside CGAL package is *not* being used, since this would br problematic.
+# (`eigen` package comes with its own FindEigen3.cmake)
+find_package(Eigen3 3.3 REQUIRED NO_MODULE)
```

Comment:
If you specify `NO_MODULE`, there is no way any `FindEigen3.cmake` will be search for (would it be in CGAL or Eigen). Instead, the CMake command `find_package` searches for `Eigen3Config.cmake`.